src/main/java/com/flightstats/hub/dao/aws/ClusterContentService.java

```diff
@@ -63,37 +64,41 @@
 
     private final static Logger logger = LoggerFactory.getLogger(ClusterContentService.class);
     private static final String CHANNEL_LATEST_UPDATED = "/ChannelLatestUpdated/";
-    private static final long largePayload = HubProperties.getLargePayload();
     private static final int queryMergeMaxWaitMinutes = HubProperties.getProperty("query.merge.max.wait.minutes", 2);
     private static final ExecutorService executorService = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("ClusterContentService-%d").build());
     private final boolean dropSomeWrites = HubProperties.getProperty("s3.dropSomeWrites", false);
-    @Inject
-    @Named(ContentDao.WRITE_CACHE)
     private ContentDao spokeWriteContentDao;
-    @Inject
-    @Named(ContentDao.SINGLE_LONG_TERM)
     private ContentDao s3SingleContentDao;
-    @Inject
-    @Named(ContentDao.READ_CACHE)
     private ContentDao spokeReadContentDao;
-    @Inject
-    @Named(ContentDao.BATCH_LONG_TERM)
     private ContentDao s3BatchContentDao;
-    @Inject
-    @Named(ContentDao.LARGE_PAYLOAD)
     private ContentDao s3LargePayloadContentDao;
-    @Inject
+    private S3WriteQueue s3WriteQueue;
     private ChannelService channelService;
-    @Inject
     private LastContentPath lastContentPath;
-    @Inject
-    private S3WriteQueue s3WriteQueue;
-    @Inject
     private HubUtils hubUtils;
 
-    public ClusterContentService() {
+    @Inject
+    public ClusterContentService(
+                            ChannelService channelService,
+                            @Named(ContentDao.WRITE_CACHE) ContentDao spokeWriteContentDao,
+                            @Named(ContentDao.READ_CACHE) ContentDao spokeReadContentDao,
+                            @Named(ContentDao.SINGLE_LONG_TERM) ContentDao s3SingleContentDao,
+                            @Named(ContentDao.LARGE_PAYLOAD) ContentDao s3LargePayloadContentDao,
+                            @Named(ContentDao.BATCH_LONG_TERM) ContentDao s3BatchContentDao,
+                            S3WriteQueue s3WriteQueue,
+                            LastContentPath lastContentPath,
+                            HubUtils hubUtils) {
```

Comment:
Nice. Constructor based injection is more readable and definitely a better choice since it throws exception sooner if one of the dependencies is not available.